### PR TITLE
Commit sha for executions

### PR DESCRIPTION
**Pull request by juliaromare**
When getting workflow instances through the API, expose commit sha of execution, if available.

### CODE REVIEW

**Comment by juliaromare** [comment]
ptal @NatashaL @danielnorberg

**Comment by NatashaL** [comment]
looks good to me 👍

File: styx-common/src/main/java/com/spotify/styx/model/data/WFIExecutionBuilder.java [hunk 1]
```diff
@@ -123,6 +127,7 @@ public Void submit(WorkflowInstance workflowInstance, ExecutionDescription execu
         String executionId) {
       currWorkflowInstance = workflowInstance;
       currDockerImg = executionDescription.dockerImage();
+      currCommitSha = executionDescription.commitSha().get();
```
Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/model/data/WFIExecutionBuilder.java]: This crashes for instances with no `commitSha`

File: styx-common/src/main/java/com/spotify/styx/model/EventVisitor.java [hunk 2]
```diff
@@ -54,7 +54,7 @@ R submit(@Getter WorkflowInstance workflowInstance, ExecutionDescription executi
   @Deprecated
   R timeTrigger(@Getter WorkflowInstance workflowInstance);
   @Deprecated
-  R created(@Getter WorkflowInstance workflowInstance, String executionId, String dockerImage);
+  R created(@Getter WorkflowInstance workflowInstance, String executionId, String dockerImage, String commitSha);
```
Comment (danielnorberg) [inline styx-common/src/main/java/com/spotify/styx/model/EventVisitor.java]: This will crash when reading old `created` events.

**Pull request merged**